app/src/main/java/org/oppia/app/player/state/StatePlayerRecyclerViewAssembler.kt

```diff
@@ -1325,6 +1340,7 @@ class StatePlayerRecyclerViewAssembler private constructor(
           // becomes null such as in the case of the solution becoming available).
           (fragment as ShowHintAvailabilityListener).onHintAvailable(helpIndexToShow)
           previousHelpIndex = helpIndexToShow
+          ishintVisibile = true
```

Comment:
I think mechanically this solution looks good, but could we possibly leverage the existing ``ShowHintAvailabilityListener``? I think calling ``onHintAvailable`` with a default instance of ``HelpIndex`` will lead to the bulb being hidden. Unless I'm mistaken, it might simplify the current solution quite a bit.

Reply:
Yes, the default instance of `HelpIndex` does hide the bulb icon, but we need to show it again if a user comes back to the current state where the bulb was previously visible. 

1. Hint bulb visible either by submitting 2 wrong answers or after a period of time
2. Go to the previous state which is a completed state (hide hint bulb)
3. Come back to the previous state which was the last state where hint bulb is visible
4. Make hint bulb visible again

Reply:
In order to make it visible again, we have to trigger bulb visibility which statefragment and questionplayer view model are handling.

Reply:
Can we call maybeScheduleShowHint in the assembler when switching from a completed to a pending state?

Reply:
Yes, it is already calling and that's where I am doing the checking if the bulb icon was previously visible or not. 

```
 if (ishintVisibile) {
        (fragment as ShowHintAvailabilityListener).hintVisibilityBetweenState(
          /* hintVisibility= */ true
        )
      }
```

Reply:
Done